pkg/controllers/awsloadbalancercontroller/deployment_test.go

```diff
@@ -686,17 +585,239 @@ func TestEnsureDeployment(t *testing.T) {
+						Sources: []corev1.VolumeProjection{{
+							ServiceAccountToken: &corev1.ServiceAccountTokenProjection{
+								Audience:          "openshift",
+								ExpirationSeconds: pointer.Int64(3600),
+								Path:              "token",
+							},
+						}},
+					}}},
+					corev1.Volume{Name: "trusted-ca", VolumeSource: corev1.VolumeSource{ConfigMap: &corev1.ConfigMapVolumeSource{LocalObjectReference: corev1.LocalObjectReference{Name: "test-trusted-ca"}}}},
+				).build(),
+			},
+			trustedCAConfigMap: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-trusted-ca",
+					Namespace: "test-namespace",
+				},
+				Data: map[string]string{
+					"ca-bundle.crt": "--ca bundle--",
+				},
+			},
+			expectedDeployment: testDeployment(
+				"cluster",
+				"test-namespace",
+				"test-sa",
+				"test-serving",
+			).withTemplateAnnotation("networking.olm.openshift.io/trusted-ca-configmap-hash", "745c799ac442a05bf2ff97c75089850a00471d9dd51392ef9ff9f6ea610d1071").
+				withContainers(
+					testContainer("controller", "test-image").withDefaultEnvs().withVolumeMounts(
+						corev1.VolumeMount{Name: "aws-credentials", MountPath: "/aws"},
+						corev1.VolumeMount{Name: "tls", MountPath: "/tls"},
+						corev1.VolumeMount{Name: "bound-sa-token", MountPath: "/var/run/secrets/openshift/serviceaccount", ReadOnly: true},
+						corev1.VolumeMount{Name: "trusted-ca", MountPath: "/etc/pki/tls/certs/albo-tls-ca-bundle.crt", SubPath: "ca-bundle.crt"},
+					).withSecurityContext(corev1.SecurityContext{
+						Capabilities:             &corev1.Capabilities{Drop: []corev1.Capability{"ALL"}},
+						Privileged:               pointer.BoolPtr(false),
+						RunAsNonRoot:             pointer.BoolPtr(true),
+						AllowPrivilegeEscalation: pointer.BoolPtr(false),
+						SeccompProfile:           &corev1.SeccompProfile{Type: corev1.SeccompProfileTypeRuntimeDefault},
+					}).build(),
+				).withResourceVersion("3").withVolumes(
+				corev1.Volume{Name: "aws-credentials", VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: "test-credentials"}}},
+				corev1.Volume{Name: "tls", VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: "test-serving"}}},
+				corev1.Volume{Name: "bound-sa-token", VolumeSource: corev1.VolumeSource{Projected: &corev1.ProjectedVolumeSource{
+					DefaultMode: pointer.Int32(420),
+					Sources: []corev1.VolumeProjection{{
+						ServiceAccountToken: &corev1.ServiceAccountTokenProjection{
+							Audience:          "openshift",
+							ExpirationSeconds: pointer.Int64(3600),
+							Path:              "token",
+						},
+					}},
+				}}},
+				corev1.Volume{Name: "trusted-ca", VolumeSource: corev1.VolumeSource{ConfigMap: &corev1.ConfigMapVolumeSource{LocalObjectReference: corev1.LocalObjectReference{Name: "test-trusted-ca"}}}},
+			).build(),
+		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
+			for k, v := range tc.existingEnvVars {
```

Comment:
This only runs for one test case.  In other words, only one of your test cases contains `existingEnvVars`.  Seems like it would be better in its own test instead of here.

Reply:
Done.